### ISSUE TITLE
Incorrect Yearn v2 USDC price

### ISSUE THREAD

**Issue by e-gons** [rotki/rotki#3355]
## Problem Definition
The Yearn v2 USDC vault shows an incorrect price of 1. This is the result of the issue noted by the devs here: https://github.com/rotki/rotki/blob/975f97deea470dd5f659f9928373ee134bfe54a4/rotkehlchen/chain/ethereum/defi/price.py#L144

## Potential Solution
In rotkehlchen/chain/ethereum/defi/price.py -> `handle_underlying_price_yearn_vault`

Instead of:
```python
    # or understand why yUSDC and yUSDT which have 6 decimals don't work correctly
    div_decimals = 18
    price_per_full_share = contract.call(ethereum, 'getPricePerFullShare')
    usd_value = FVal(asset_price * price_per_full_share) / 10 ** div_decimals
    return usd_value
```

Try:
```python
    # see https://github.com/BobTheBuidler/ypricemagic/blob/882aa2071a918937e77e0b85e5f52191a4714d28/ypricemagic/yearn.py#L13
    # Yearn v1 vaults use getPricePerFullShare scaled to 18 decimals
    # Yearn v2 vaults use pricePerShare scaled to underlying token decimals

    if hasattr(contract, "pricePerShare"):
        div_decimals = contract.call(ethereum, "decimals")
        price_per_share = contract.call(ethereum, "pricePerShare")
        usd_value = FVal(asset_price * price_per_share) / 10 ** div_decimals
        return usd_value

    if hasattr(contract, "getPricePerFullShare"):
        div_decimals = 18
        price_per_full_share = contract.call(ethereum, "getPricePerFullShare")
        usd_value = FVal(asset_price * price_per_full_share) / 10 ** div_decimals
        return usd_value

    if hasattr(contract, "getPricePerShare"):
        div_decimals = contract.call(ethereum, "decimals")
        price_per_share = contract.call(ethereum, "getPricePerShare")
        usd_value = FVal(asset_price * price_per_share) / 10 ** div_decimals
        return usd_value
```

### System Description

Operating system: macOS Monterey
Rotki version: 1.20.0

**Comment by yabirgb** [comment]
@e-gons Thank you for explaining the issue with such detail citing the yearn code. Will review your suggestions and fix the issue hopefully in the next pach release :)

**Comment by LefterisJP** [comment]
Should be fixed by #3371

**Comment by e-gons** [comment]
Fixed! Thanks again, team.
